src/app.ts

```diff
@@ -7,10 +28,46 @@
 	//      https://github.com/nodejs/node/pull/17793
 	const dns = require('dns');
 	const { lookup } = dns;
-	dns.lookup = (name: string, opts: any, cb: (err?: Error) => void) => {
+
+	dns.lookup = (
+		name: string,
+		opts: Partial<DnsLookupOpts>,
+		cb: DnsLookupCallback,
+	) => {
 		if (typeof cb !== 'function') {
 			return lookup(name, { verbatim: true }, opts);
 		}
+
+		if (name.endsWith('.local')) {
+			return Promise.all([
+				mdnsResolver.resolve4(name).catch(() => {
+					return '';
+				}),
+				mdnsResolver.resolve6(name).catch(() => {
+					return '';
+				}),
+			]).then(([v4, v6]) => {
+				const allAddresses = [v4, v6].filter((addr) => addr !== '');
```

Comment:
Shouldn't we avoid doing a lookup for both addresses if we aren't going to need it?

Reply:
Agreed. I have remedied this.